From b276609d15287faba0f104c93d9b8c3058801f78 Mon Sep 17 00:00:00 2001
From: lingsheng <lingsheng@huawei.com>
Date: Mon, 14 Sep 2020 20:48:03 +0800
Subject: [PATCH] fix gcc version check

---
 CONFIG/src/atlconf_misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CONFIG/src/atlconf_misc.c b/CONFIG/src/atlconf_misc.c
index dcf6a99..1c90105 100644
--- a/CONFIG/src/atlconf_misc.c
+++ b/CONFIG/src/atlconf_misc.c
@@ -848,7 +848,7 @@ int CompIsGcc(char *comp)
 	  strstr(res, "(HCC)")  || strstr(res, " HCC ") ||strstr(res, "(HCC ") ||
           strstr(res, "GNU Fortran") ||
           (res[0] == 'g' && res[1] == 'c' && res[2] == 'c' &&
-           (res[3] == ' ' || res[3] == '-'))
+           (res[3] == ' ' || res[3] == '-' || res[3] == '_'))
          )
       {
          free(res);
-- 
2.23.0

